.github/workflows/preprod-deployment.yml

```diff
@@ -164,4 +165,4 @@ jobs:
           token: "${{ github.token }}"
           target_url: https://downloads.spacelift.dev/spacelift-vcs-agent
           state: "failure"
-          deployment_id: ${{ steps.deployment.outputs.deployment_id }}
+          deployment_id: ${{ needs.preprod-agent-deployment.outputs.deployment_id }}
```

Comment:
Just wondering, if the build job fails, will we mark the deployment as failed? If not, I think we've got two options:

1. (the simple option) Move the "Create GitHub deployment" step into the upload job.
2. Add a failure task to the build job as well.

I think here we could get away with option 1 on the grounds that you could argue the deployment hasn't started until we begin to upload the binary/image, but honestly I think either is fine here.

Reply:
I think I can just add an extra failure step (like the one in the upload job) to the build job as well. WDYT?

Reply:
Yeah - that should be fine.

Reply:
Added the extra step to both preprod and prod deployment workflows.